src/main/java/microservices/userServiceBasic/api/model/UserDetails.java

```diff
@@ -0,0 +1,41 @@
+package microservices.userServiceBasic.api.model;
+
+import java.util.UUID;
+
+public class UserDetails {
+
+	private final UUID id;
+	private final String firstName;
+	private final String lastName;
+	private final String email;
+
+	public String getFirstName() {
+		return firstName;
+	}
+
+	public String getLastName() {
+		return lastName;
+	}
+
+	public String getEmail() {
+		return email;
+	}
+
+	public UserDetails(UUID id, String firstName, String lastName, String email) {
+		this.id = id;
+		this.firstName = firstName;
+		this.lastName = lastName;
+		this.email = email;
+	}
+
+	public UserDetails(String firstName, String lastName, String email) {
+		this.id = new UUID(0L, 0L);
+		this.firstName = firstName;
+		this.lastName = lastName;
+		this.email = email;
+	}
+
+	public UUID getId() {
```

Comment:
It's good practice to put all the getters (and setters) in the same place. Put this one before the constructors.